### DISCUSSION TITLE
Package release cadence e.g. speed up

### DISCUSSION THREAD

**Discussion by TimNZ** [nextauthjs/next-auth#1033] [Help]
I genuinely appreciate the time everyone puts into open-source things,
but I spent an hour debugging a bug into the Prisma adapter which has had a fix committed since Sep 1st and still not released, and that is annoying.

What can I do to help speed things up?

**Comment by balazsorban44** [comment]
Could you link to the PR containing that commit? All PRs merged into the canary branch should be released immediately as we now have set up automatic releases.

Please also keep in mind that the maintainers have their own life, and sometimes (either because of work or personal life/family) they need a pause from their hobbies (like this project). Even though it is "annoying", it is just the way Open Source is.

You can always fork the repo and create your own npm release with your implemented fixes, if speed is a problem.

**Reply by TimNZ** [reply]
Ignore me, dumb moment.
I didn't check what versions were available beyond the numbered release, and see all the canary packages.
A lot of packages have a 'canary' for the last build, which is what I tried last week.

**Comment by iaincollins** [comment]
I've been really fallen behind with PR requests since 3.1 and had to take a break, sorry about that.

@balazsorban44 has done an amazing job recently - including providing us with automatic Canary Releases and I'm incredibly grateful.

It might be a little while before we cut the canary release into the main branch but am confident we can get to that a lot sooner now we have canary releases to take the pressure off of QA/testing cycles!

Longer term I think we'd want to split off the Adapters from core to make it easier to iterate on them; that was always my intent so that people can create and maintain their own Adapaters for different databases (e.g. I see there is one for DynamoDB on NPM now).

In this case you might even want to fork the Adapter code into your own repo and import it directly - if we change the Adapter API it would be breaking change like 4.0, and TBH I don't see us changing the actual API for them much if at all; so it should be safe to run your own custom Adapter and that's very much intended to be a pattern by design.

**Reply by TimNZ** [reply]
Hey @iaincollins 
Thanks for the reply.
Appreciate the time you've put into putting free things into the world for our benefit!

As I mentioned above, I had done a quick try of next-auth@canary or next-auth@latest last week and found no package, and didn't check for what packages had been published.  
This is my ignorance of package versioning conventions, and not doing a check for later published betas.

I did end up creating my own adapter which works perfectly.

Thanks again.
